src/librustc_metadata/diagnostics.rs

```diff
@@ -91,6 +91,204 @@ You need to link your code to the relevant crate in order to be able to use it
 well, and you link to them the same way.
 "##,
 
+E0466: r##"
+Invalid macro import declarations.
+
+Causes of this error:
+
+```ignore
+#[macro_use(a_macro(another_macro))]    // error: invalid import declaration
+extern crate some_crate;
+
+#[macro_use(i_want = "some_macros")]    // error: invalid import declaration
+extern crate another_crate;
+```
+
+This is a syntax error at the level of attribute declarations.
+
+The proper syntax for macro imports is the following:
+
+```ignore
+// // some_crate contains:
+// #[macro_export]
+// macro_rules! get_tacos {
+//     ...
+// }
+//
+// #[macro_export]
+// macro_rules! bring_beer {
+//     ...
+// }
+#[macro_use(get_tacos, bring_beer)]     // imports macros get_tacos and
+extern crate some_crate;                // bring_beer from some_crate
+```
+
+If you would like to import all exported macros, write `macro_use` with no
+arguments.
+"##,
+
+E0467: r##"
+Invalid or no macros listed for reexport.
+
+Causes of this error:
+
+```ignore
+#[macro_reexport]                   // error: no macros listed for export
+extern crate macros_for_good;
+```
+```ignore
+#[macro_reexport(fun_macro = "foo")] // error: not a macro identifier
+extern crate macros_for_good;
+```
+
+This is a syntax error at the level of attribute declarations.
+
+Currently, `macro_reexport` requires at least one macro name to be listed.
+Unlike `macro_use`, listing no names does not reexport all macros from the
+given crate.
+
+Decide which macros you would like to export and list them properly.
+
+These are proper reexport declarations:
+
+```ignore
+#[macro_reexport(some_macro, another_macro)]
+extern crate macros_for_good;
+```
+"##,
+
+E0468: r##"
+A non-root module attempts to import macros from another crate.
+
+Example of erroneous code:
```

Comment:
The point of adding `compile_fail,<errno>` is to avoid out of date error explanations like it happened a lot. If the error isn't thrown by the code anymore, then it means it has changed and the error explanation should be updated.